Improve error handling in solita.etp.api.energiatodistus-xml/post

muep opened the pull request:
This addresses AE-1078

hkorpi commented on etp-backend/src/main/clj/solita/etp/api/energiatodistus_xml.clj:
```diff
@@ -297,12 +298,22 @@
+                     :invalid-value response/bad-request
+                     :missing-value response/bad-request
+                     :schema-tools.coerce/error response/bad-request
+                     :unique-violation response/bad-request
```
unique violationin voi tiputtaa pois

hkorpi replied:
se on tässä tapauksessa aina järjestelmä virhe 500 jos esiintyy

muep replied:
uusimmassa versiossa on poistettu

MattiNieminen commented on etp-backend/src/main/clj/solita/etp/api/energiatodistus_xml.clj:
```diff
@@ -286,7 +287,7 @@
    xml/emit-str))
 
-(defn success-response [id warnings]
+(defn success-body [id warnings]
```
Nimen muutos ehkä vähän hämäävä, koska nämä funktiot palauttavat ihan oikean response-mapin, eivätkä pelkkää bodyä.

muep replied:
Tuo `response` -funktion `xml/emit-str` kuulostaa nimenä siltä että oltaisiin jo vähintään tekstipötkönä.

hkorpi replied:
eikös tämä tuota juurikin bodyn josta sitten tehdään response r/reponse funktiolla?

muep replied:
Nimesin nämä nyt soap-bodyiksi kun oli semmoista puhetta aamupäivällä

hkorpi replied:
nämä piti pitää ennallaan, mutta se response funktio piti nimetä soap-body:ksi

muep replied:
ah joo, minulla taas meni viime muutoskerralla tuo saman tiedoston `response` ihan ohi.

hkorpi replied:
nämä sinun muuttamat oli siis ihan oikein nämä tuottavat bodyn, mutta toi response funktio on väärin nimetty, koska se oikeasti tuottaa siis soap-bodyn

muep replied:
Nyt on nimetty `response`  -> `soap-body`

muep committed:
AE-1078 Improve error handling

This improves error handling in
solita.etp.api.energiatodistus-xml/post

On exceptions that are caught at the XML ET reception, one if these
actions is taken:
- a 400-series status response with an XML-formatted body is produced
- the exception is re-thrown so that we get an usual backtrace. The
  content of the response is not kept in XML format in this case
